integration-tests/test_unregister.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2024 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+
+
+import pytest
+from utils import loop_until
+from constants import RHSM, RHSM_UNREGISTER
+
+import logging
+from functools import partial
+from funcy import first
+from dasbus.error import DBusError
+
+logger = logging.getLogger(__name__)
+
+locale = "en_US.UTF-8"
+
+# Tests describe a case when an application unregisters a system.
+#
+# The API should handle even wrong cases friendly
+# - ie. when a system is not registered the API should provide usefull feedback
+
+
+def test_unregister(any_candlepin, subman, test_config):
+    """
+    https://www.candlepinproject.org/docs/subscription-manager/dbus_objects.html#unregister
+    """
+    candlepin_config = partial(test_config.get, "candlepin")
+    subman.register(
+        username=candlepin_config("username"),
+        password=candlepin_config("password"),
+        org=candlepin_config("org"),
+        environments=first(candlepin_config("environment", "names")),
```

Comment:
It is not necessary to use environment here. It is reason, why this test fails.

Reply:
removed